File: src/main/resources/templates/swift/xcode/solution/${appName}/Sort/BubbleSort.swift
//
//  BubbleSort.swift
//  ${appName}
//

import Foundation

public class BubbleSort: SortStrategy {
    /**
     Sorts dates with BubbleSort.

     - Parameter input: the List of Dates to be sorted
    */
    
    /*
     Task 1.1:
     Implement the Bubble Sort algorithm
    */
    public func performSort(_ input: [Date]) -> [Date] {
        var sortedInput = input
        for i in (0 ..< sortedInput.count).reversed() {
            for j in 0 ..< i where sortedInput[j] > sortedInput[j + 1] {
                    let temp: Date! = sortedInput[j]
                    sortedInput[j] = sortedInput[j + 1]
                    sortedInput[j + 1] = temp
            }
        }
        return sortedInput
    }
}

File: src/main/resources/templates/go/solution/bubblesort.go
package ${packageName}

import "time"

type BubbleSort struct{}

func NewBubbleSort() *BubbleSort {
	return new(BubbleSort)
}

func (b *BubbleSort) PerformSort(input []time.Time) {
	for i := len(input) - 1; i >= 0; i-- {
		for j := 0; j < i; j++ {
			if input[j].After(input[j+1]) {
				input[j], input[j+1] = input[j+1], input[j]
			}
		}
	}
}
